apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/OpenApiDocs.java

```diff
@@ -183,4 +190,70 @@ private void setRequestAuth() {
     requestAuth =
         new RequestSpecBuilder().setBaseUri(baseServerUrl).setAuth(testCertificate).build();
   }
+
+  /**
+   * Adds header parameters to all paths except '/metadata'.
+   *
+   * @param specStr the OpenAPI spec String
+   * @param apiVersion the bfd api version, either V1 or V2.
+   * @return new OpenAPI spec String with header parameters added.
+   */
+  private String addHeaderParameters(String specStr, String apiVersion) {
+    Yaml yaml = new Yaml();
+    Map<String, Object> spec = yaml.load(specStr);
+
+    InputStream headersYaml =
+        this.getClass().getClassLoader().getResourceAsStream("openapi/headers.yaml");
+    Map<String, Object> headers = yaml.load(headersYaml);
+
+    // add header params to be used in refs
+    var components = (Map<String, Object>) spec.get("components");
+    var headerComponents = (Map<String, Object>) headers.get("components");
+    var headerParameters = (Map<String, Object>) headerComponents.get("parameters");
```

Comment:
I like the use of `var` here. Since a cast is needed anyway specifying the variable type explicitly would have just added redundancy.